app/ide-desktop/lib/content/src/utils.ts

```diff
@@ -0,0 +1,30 @@
+/** @file Module containing utility functions used throughout this module,
+ * but don't fit anywhere else. */
+
+/** Used to create a "branded type", which is a type intersected with a `Brand<'Name'>`.
+ *
+ * This is useful in parameters that require values from a certain source,
+ * for example IDs for a specific object type.
+ *
+ * It is similar to a `newtype` in other languages,
+ * however because TypeScript is structurally typed, a branded type is assignable to its base type:
+ * `a: string = b as (string & Brand<'Name'>)` is valid. */
+export interface Brand<T extends string> {
+    $brand: T
+}
+
```

Comment:
hmm, maybee this should be named `Newtype` then because this name would be recognizable immediately by people knowing this concept from other languages? TBH, I did not understand what is Brand until I've read these docs, and if it was named `Newtype`, I would understand that immediately by reading the usage place.

Reply:
ah, that's a very good idea